tools/ci_build/github/linux/docker/scripts/install_ubuntu.sh

```diff
@@ -111,7 +111,10 @@ fi
 
 if [ "$OS_VERSION" = "16.04" ]; then
     if [ "$PYTHON_VER" != "3.5" ]; then
+        # https://stackoverflow.com/questions/43621584/why-cant-i-install-python3-6-dev-on-ubuntu16-04
+        apt-get install software-properties-common
         add-apt-repository -y ppa:deadsnakes/ppa
```

Comment:
I prefer to use ubuntu 18 instead. Because such code is hard to maintain. 
1. The ppa is totally out of our control.  nobody knows if it is safe to use.
2. You'll have two pythons in the docker image. pip doesn't work as you expected, you have to do it like: "python3.6 -m pip install ... " So. maybe a few scripts need be changed also.

Reply:
So do you (@snnn ) mean  in onnxruntime/tools/ci_build/github/linux/docker/Dockerfile.ubuntu_gpu we can change:
FROM nvidia/cuda:10.1-cudnn7-devel-ubuntu16.04 
to 
FROM nvidia/cuda:10.1-cudnn7-devel-ubuntu18.04 

@kit1980 was making this related change which brought some concerns for using ubuntu18 in this PR:  
https://github.com/microsoft/onnxruntime/pull/3231

It is great if we can agree on replace ubuntu16 with ubuntu18. 

@snnn , I will undo change here. Please comment on PR: https://github.com/microsoft/onnxruntime/pull/3231 so that we can conclude the way to run Python3.6.

Reply:
We have a ubuntu 18 image already, tensorrt is using it. The PR #3231 changed too much, would it be possible that just create a new Dockerfile and a new pipeline?

Reply:
@snnn In the latest iteration of #3231 only the Python version for Ubuntu 16.04 changed from 3.5 to 3.6. The Ubuntu 16.04 is still used, no changes there now.

Reply:
>     1. The ppa is totally out of our control.  nobody knows if it is safe to use.

The ppa is already in the script. The change is to add `sudo apt-get update` after already present `add-apt-repository -y ppa:deadsnakes/ppa`, which is a fix to an obvious typo/bug (you have to do update after adding a repo.)

Reply:
This PR is to create a new pipeline. Its purpose is to run nightly on more powerful (hosted) agents to test training features that are not supported by the current CI agents (for example mixed precision training). Some tests may take long to complete.

#3231 is to enable running frontend unit tests (which requires Python 3.6) that are needed for every PR.

By "create a new Dockerfile and a new pipeline" do you mean to run CI for every PR with both Ubuntu16 (with Python3.5) and Ubuntu18 (with Python3.6)? This mean to consume more CI resources.

Reply:
> tensorrt

Dockerfile.tensorrt relies on pre-installed conda to use python (/opt/miniconda/), so Ubuntu 18.04 does not play an important role for the python availability.

16.04 seems to be officially supported, as Dockerfile.source and others Dockerfiles relies on it, so it should be OK to depend on it for this pipeline. Whether we should use PPA or not is a different discussion, but this PR doesn't introduce it.

I agree that we should upgrade Ubuntu to at least 18.04, but that it should be a larger (and separate) task that encompasses all build and test pipelines, not just this one.